### ISSUE TITLE
Service Account token for Pod IAM role is misconfigured with incorrect OIDC provider CA cert thumbprint

### ISSUE THREAD

**Issue by metral** [pulumi/pulumi-eks#342]
### Problem description

The projected service account token used to assume an IAM role from a Pod is not properly configured with the correct OIDC provider root CA thumbprint.

There's a mismatch in the thumbprint expected by AWS when issuing the temporary credentials for the Pod

### Errors & Logs

```
time="2020-03-04T02:57:11Z" level=error msg="WebIdentityErr: failed to retrieve credentials\ncaused by: 
InvalidIdentityToken: OpenIDConnect provider's HTTPS certificate doesn't match configured thumbprint\n\tstatus code: 400, 
request id: 8586235f-772e-4e0b-ba22-00b99de7905e"
```

### Affected product version(s)

v0.18.22

**Comment by metral** [comment]
Re-opening as we're sporadically seeing this in https://github.com/pulumi/pulumi-eks/issues/349

**Comment by metral** [comment]
Closing as the comment above was on an older, non-patched eks version (v0.18.22), and v0.18.23 and v0.18.24 had the updated fixes.
